SYCL/ESIMD/api/functional/ctors/ctor_copy.cpp

```diff
@@ -47,11 +47,11 @@ struct var_decl {
   static std::string get_description() { return "variable declaration"; }
 
   template <typename DataT, int NumElems>
-  static simd<DataT, NumElems> call_simd_ctor(const DataT *ref_data) {
+  static void call_simd_ctor(const DataT *const ref_data, DataT *const out) {
     simd<DataT, NumElems> source_simd;
     source_simd.copy_from(ref_data);
     simd<DataT, NumElems> simd_by_var_decl{source_simd};
```

Comment:
```suggestion
    simd<DataT, NumElems> simd_by_var_decl(source_simd);
```
We should ensure we are calling exactly the constructor required.

Reply:
I've applied this comment in this commit: a6fac3bb1b02498714bd0edbdd9648053e8a27f7